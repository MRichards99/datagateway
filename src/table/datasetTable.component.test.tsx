import React from 'react';
import { shallow as enzymeShallow } from 'enzyme';
import createShallow from '@material-ui/core/test-utils/createShallow';
import DatasetTable from './datasetTable.component';

describe('DatasetTable component', () => {
  let shallow: typeof enzymeShallow;

  beforeEach(() => {
    shallow = createShallow({});
  });

  it('should render correctly', () => {
    const wrapper = shallow(<DatasetTable />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render details panel correctly', () => {
    const row = {
      name: 'Test',
      size: 1,
      createTime: new Date('2019-06-10 00:00:00'),
      modTime: new Date('2019-06-10 00:00:00'),
    };
    const wrapper = shallow(<DatasetTable />);

    const detailPanel = wrapper.prop('detailPanel')(row);
    const detailPanelWrapper = shallow(detailPanel);

    expect(detailPanelWrapper).toMatchSnapshot();
  });
});
