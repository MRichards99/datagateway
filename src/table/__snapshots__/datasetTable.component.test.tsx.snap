// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dataset table component renders Dataset title as a link 1`] = `
<Link
  to="/browse/investigation/1/dataset/1/datafile"
>
  <a
    href="/browse/investigation/1/dataset/1/datafile"
    onClick={[Function]}
  >
    Test 1
  </a>
</Link>
`;

exports[`Dataset table component renders correctly 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
  style={
    Object {
      "height": 768,
      "width": "100%",
    }
  }
>
  <WithStyles(VirtualizedTable)
    columns={
      Array [
        Object {
          "cellContentRenderer": [Function],
          "dataKey": "NAME",
          "filterComponent": [Function],
          "label": "Name",
        },
        Object {
          "cellContentRenderer": [Function],
          "dataKey": "SIZE",
          "label": "Size",
        },
        Object {
          "dataKey": "CREATE_TIME",
          "label": "Create Time",
        },
        Object {
          "dataKey": "MOD_TIME",
          "label": "Modified Time",
        },
      ]
    }
    data={
      Array [
        Object {
          "CREATE_TIME": "2019-07-23",
          "ID": 1,
          "INVESTIGATION_ID": 1,
          "MOD_TIME": "2019-07-23",
          "NAME": "Test 1",
          "SIZE": 1,
        },
      ]
    }
    detailsPanel={[Function]}
    onSort={[Function]}
    sort={Object {}}
  />
</div>
`;

exports[`Dataset table component renders details panel correctly 1`] = `
<div>
  <WithStyles(ForwardRef(Typography))>
    <b>
      Name: 
    </b>
     
    Test 1
  </WithStyles(ForwardRef(Typography))>
  <WithStyles(ForwardRef(Typography))>
    <b>
      Description: 
    </b>
     
    Test 1
  </WithStyles(ForwardRef(Typography))>
</div>
`;
